pkg/kube/inject/inject.go

```diff
@@ -1111,36 +1115,42 @@ func rewriteCniPodSpec(annotations map[string]string, spec *SidecarInjectionSpec
 
 // overwriteClusterInfo updates cluster name and network from url path
 // This is needed when webconfig config runs on a different cluster than webhook
+// follow format: /inject/k1/v1/k2/v2, any kv order works
+// eg. "/inject/cluster/cluster1", "/inject/net/network1/cluster/cluster1"
 func overwriteClusterInfo(containers []corev1.Container, path string) {
 	res := strings.Split(path, "/")
-	if len(res) >= 5 {
-		// if len is less than 5, not enough length for /cluster/X/net/Y
-		clusterName, clusterNetwork := "", ""
-		clusterName = res[len(res)-3]
-		clusterNetwork = res[len(res)-1]
+	newEnvs := make(map[string]string)
 
-		log.Debugf("Updating cluster info based on clusterName: %s clusterNetwork: %s\n", clusterName, clusterNetwork)
+	for i := 2; i+1 < len(res); i += 2 { // skip '/inject'
+		k := res[i]
+		v := res[i+1]
+		if URLParameterToEnv[k] != "" {
```

Comment:
@linsun @GregHanson just curious, was there a reason why we didn't just use URL parameters instead decoding the path? If we used parameters, I suspect we could leverage built-in libraries for parsing it.

Reply:
Webhooks doesn't allow them :frowning_face: